docs/source/keys.rst

```diff
@@ -0,0 +1,293 @@
+- ``sig(k) of T`` generates the signature of transaction ``T`` using private key ``k``. It cannot be used inside the definition of a  transaction.
+
+
+The signing operation signs all the fields of the transaction *but*
+the witness fields.  This is necessary not to incurr in
+infinite signatures, and it allows multiple signatures to be added
+without invalidating previous ones.
+   
+Consider the following transaction ``A_funds``, redeemable with 
+a signature made by Alice.
+
+.. code-block:: btm
+		
+    // Alice's public key
+    const kApub = pubkey:033806fe039c8d149c8e68f4665b4a479acab773b20bddf7df0df59ba4f0567341
+
+    // tx redeemable with Alice's private key
+    transaction A_funds {input = _ output = 1BTC: fun(x). versig(kApub; x)}
+
+
+Let us define transaction ``TA``   to redeem transaction ``A_funds``, like this:
+   
+.. code-block:: btm
+
+	//Alice's private key	
+	const kA = key:cSFqKAaCUGHZoGDrcreo3saZpMv9NvcVmRZVbVddbodEuzWNCDNt
+
+	// defining  transaction TA
+	transaction TA {
+		input = A_funds : sig(kA)   //signature of the transaction
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+Alternatively, it is possible to use the other construct,  like this:
+
+.. code-block:: btm
+
+	//Alice's private key	
+	const kA = key:cSFqKAaCUGHZoGDrcreo3saZpMv9NvcVmRZVbVddbodEuzWNCDNt
+
+	// transaction with witness not specified
+	transaction T {
+		input = A_funds : _ 
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+	// signing transaction T
+	const sigA = sig(kA) of T 
+
+	// defining the actual transaction TA,
+	//with the same fields as in T so to use  signature sigA
+	transaction TA {
+		input = A_funds : sigA 
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+
+Transaction ``TA`` uses as witness the signature obtained by ``T``,
+which has same input and output fields.
+
+Something equivalent can be written with the use of parametric transaction ``T_template``:
+
+.. code-block:: btm
+
+	//Alice's private key	
+	const kA = key:cSFqKAaCUGHZoGDrcreo3saZpMv9NvcVmRZVbVddbodEuzWNCDNt
+
+	// template for a parametric transaction
+	transaction T_template(s:signature) {
+		input = A_funds : s 
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+	// signing transaction T_template, without providing an argument
+	const sigA = sig(kA) of T_template(_) 
+
+	//instanciating T_template with the needed argument
+	const TA  =  T_template(sigA)
+
+
+Transaction ``T_template`` is parameteric and asks for a signature to
+be used as witness for ``A_funds``. That signature can only be the
+signature of ``T_template`` itself. Hence, after having generated
+``sigA``, it is inserted into ``T_template`` to obtain final
+transaction ``TA``.
+
+
+
+
+
+When the signatures to be included in a transaction are several, the
+protocol is the following: each participant signs a template for the
+transaction, and send that signature to someone which collects all the
+signatures together.
+
```

Comment:
Rifraserei. Non mi piace si parli di "protocol", preferirei "approach".

Metterei inoltre in evidenza che l'obiettivo è non condividere la chiave privata tra di loro.